pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -1864,7 +1864,8 @@ statfunc u32 cgroup_skb_generic(struct __sk_buff *ctx, enum flow_direction flow_
 
         // For older kernels, also check the existing socket map.
         if (LINUX_KERNEL_VERSION < KERNEL_VERSION(5, 11, 0)) {
-            struct sock *key = (void *) ctx->sk;
+            struct sock *key = (void *) BPF_CORE_READ(ctx, sk);
```

Comment:
I think what we could try is to either use `sk` directly in the map lookup and if this doesn't work, try reading the `sk` pointer from the `sk_buff` struct (not `__sk_buff`).

Reply:
The lookup can be fixed via 
```c
            void *key = sk;
            existing_netctx = bpf_map_lookup_elem(&existing_sockets_map, &key);
```

It is required to assign a new variable though, as otherwise the verifier thinks `sk` is invalid further down the line.